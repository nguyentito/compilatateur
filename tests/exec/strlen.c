
int strlen(char *s) {
  int len;
  len = 0;
  while (*s++) len++;
  return len;
}

void print_int(int n) {
  if (n > 9) print_int(n / 10);
  putchar('0' + n%10);
}

int main() {
  print_int(strlen(""));
  putchar(10);
  print_int(strlen("hello world"));
  putchar(10);
  print_int(strlen("hello world\n"));
  putchar(10);
  print_int(strlen("foo bar"));
  putchar(10);
  print_int(strlen("answer to the question of life, the uni..."));
  putchar(10);
  return 0;
}
