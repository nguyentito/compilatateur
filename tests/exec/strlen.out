0
11
12
7
42
